### ISSUE TITLE
cursor jumps to the top after toggling editor mode

### ISSUE THREAD

**Issue by qihuazhong** [BoostIO/Boostnote#3263]
# Current behavior

I was editing the text at the bottom, toggling the editor mode (either by keyboard or mouse) will make the cursor jump to the top.

This is annoying because a lot of time I just toggle to have a quick look of the preview, and if everything is ok, toggle back to without the preview. 

# Expected behavior

Cursor in the editor to remain at the same position after toggling editor mode

# Steps to reproduce

1. Edit the text at the bottom
2. Toggle the editor mode (either by keyboard or mouse) 
3. will make the cursor jump to the top.

# Environment

Version: 0.12.1
Electron: 3.0.8
Chrome: 66.0.3359.181
Node.js: 10.2.0
V8: 6.6.346.32
OS: Darwin x64 18.7.0

**Comment by Flexo013** [comment]
Are there any images in the note? If so then this problem is fixed in the next release, see #3023

**Comment by qihuazhong** [comment]
> Are there any images in the note? If so then this problem is fixed in the next release, see #3023

No, there is no images in the note. I am aware of the image issue. So I tested without any image and the issue still exists.

**Comment by Flexo013** [comment]
When I try to reproduce this the preview mode scroll back to the top, but the cursor in de editor view (when I switch back to it) stay in the same location it was.

Could you send the note file that produces this issue for you?

**Comment by Flexo013** [comment]
This issue seems fixed in 0.13.0.

Feel free to reopen this ticket with the extra info if the issue persists.

**Comment by qihuazhong** [comment]
![Nov-01-2019 15-39-54](https://user-images.githubusercontent.com/22231207/68051688-692c2f80-fcbe-11e9-88b8-a84734adc903.gif)

Above is the screen recording for this issue.

For me, it seems to have the same issue no matter what note I have.  Here's one example note file:
[note.txt](https://github.com/BoostIO/Boostnote/files/3799387/note.txt)

Thank you.

**Comment by Flexo013** [comment]
Upon clicking to edit a note (when in Markdown view) the edit mode should be opened with the cursor as close as possible to the line where the user clicked.
